src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
@jonorossi - I am also not completely sold on the dependency model stuff(+1). The case where there were hundreds of assemblies, is this quite common on the projects you work on? I have generally broken things up using services so never had this before. If this is common for you, then we should emulate this scenario somehow using a test. The worry here is we need to change this anyway, should we be concerned about how this executes under load?